updated icatx docs (#148)
neutron-org/neutron-docs: docs/neutron/modules/interchain-txs/overview.md

> execute [RegisterInterchainAccount message]( /neutron/modules/interchain-txs/messages#msgregisterinterchainaccount) again to
> recover access to your interchain account.
> **Note** Keep in mind, new channel is created
>

## Sudo Handlers

> The interchaintxs module in neutrond configured the following way.

Reviewer: ```suggestion
The interchaintxs module in neutrond configured the following way.

```

> wasmd sudo handler wrapped with [sudolimitwrapper](../contract-manager/overview.md#sudolimitwrapper)

Reviewer: ```suggestion
Wasmd Sudo handler wrapped with [SudoLimitWrapper](../contract-manager/overview.md#sudolimitwrapper)
```

And acknoledgement packet follows the way

> `OnAcknowledgementPacket` --> [sudolimitwrapper](../contract-manager/overview.md#sudolimitwrapper) --> `wasmd sudo handler`

Reviewer: ```suggestion
`OnAcknowledgementPacket` --> [SudoLimitErapper](../contract-manager/overview.md#sudolimitwrapper) --> `Wasmd Sudo handler`
```

> Using `sudolimitwrapper` has two purposes:

Reviewer: ```suggestion
Using `SudoLimitWrapper` has two purposes:
```

> In case of error/expensive tx

Reviewer: ```suggestion
In case of error/expensive tx:
```

1. Suppress the sudo handler error itself, and mark the ibc acknowledgement packet as received and processed. Other way, the error makes relayer send an acknowledgement again and again
2. Limit the amount of gas available for sudo handler execution. Out of gas panic will later be captured by `sudolimitwrapper` and converted into an error.
3. Information about an unsuccessfully processed ack is stored in [state](../contract-manager/state.md).

## Failed interchain txs

> Not every interchaintx executes succesfully on a remote network. Some of them fails to execute with errors and then you get ibc acknowledgement with `Error` type. In this case, in order to get additional details about the transaction parameters as well as details about the error, you can use the commands `<binary> q interchain-accounts host packet-events <channel-id> <seq-id>`

Reviewer: > `<binary> q interchain-accounts host packet-events <channel-id> <seq-id>`

Let's say `<binary>` must be replaces with a binary of a remote chain

Where:

- `channel-id` is the id of the channel on the host side of the interchain accounts
- `seq-id` - seq of the ibc message received on the host

> For example `gaiad q interchain-accounts host packet-events channel-736 1` is a transaction `fund community pool from neutron unclaimed airdrop` on `cosmoshub-4` chain. Because this command is just an alias for the transaction search functionality, it searches for the transaction using the following keys `recv_packet.packet_dst_channel = <channel-id> AND recv_packet.packet_dst_port = <port> AND recv_packet. packet_sequence = <seq-id>`. The node you are accessing may not have this transaction, due to the fact that it was included in the block a long time ago and it has already been removed by the pruning procedure.

Reviewer: Is it a real tx on a mainnet? Let's add an rpc here, so a user can copypast the command and that's it

> Unfortunately, to avoid the nondeterminism associated with error test generation, the error text is severely truncated by redact down to the error code without any additional details, before being saved to the state on the host interchain account side of the module.

Reviewer: > the error text is severely truncated by redact down to the error code without any additional details

Please provide a link to the code here

> execute [RegisterInterchainAccount message]( /neutron/modules/interchain-txs/messages#msgregisterinterchainaccount) again to
> recover access to your interchain account.
> **Note** Keep in mind, new channel is created
>

## Sudo Handlers

The interchaintxs module in neutrond configured the following way.

Wasmd Sudo handler wrapped with [SudoLimitWrapper](../contract-manager/overview.md#sudolimitwrapper)

And acknoledgement packet follows the way
`OnAcknowledgementPacket` --> [SudoLimitErapper](../contract-manager/overview.md#sudolimitwrapper) --> `Wasmd Sudo handler`

Using `SudoLimitWrapper` has two purposes:

1. Suppress the sudo handler error itself, and mark the ibc acknowledgement packet as received and processed. Other way, the error makes relayer send an acknowledgement again and again. Information about an unsuccessfully processed ack is stored in [state](../contract-manager/state.md).

> 2. Limit the amount of gas available for sudo handler execution. Out of gas panic will later be captured by `sudolimitwrapper` and converted into an error.

Reviewer 2: ```suggestion
2. Limit the amount of gas available for sudo handler execution. Out of gas panic will later be captured by `SudoLimitWrapper` and converted into an error.
```

## Failed interchain txs

> Not every interchaintx executes succesfully on a remote network. Some of them fails to execute with errors and then you get ibc acknowledgement with `Error` type. In this case, in order to get additional details about the transaction parameters as well as details about the error, you can use the commands `<binary> q interchain-accounts host packet-events <channel-id> <seq-id>`
> Where:
> 

Reviewer 2: ```suggestion
Not every interchaintx executes succesfully on a remote network. Some of them fail to execute with errors and then you get ibc acknowledgement with `Error` type. In this case, in order to get additional details about the transaction parameters as well as details about the error, you can use the command:

`<binary> q interchain-accounts host packet-events <channel-id> <seq-id>`

Where:

```

added some spaces to make it more readable. otherwise it's all compressed into a single sentence:
<img width="688" alt="image" src="https://github.com/neutron-org/neutron-docs/assets/34917380/156e38d2-e1cd-4897-930a-37542d3ae8e0">

> - `binary` is a binary on the chain you are working with
> - `channel-id` is the id of the channel on the host side of the interchain accounts
> - `seq-id` - seq of the ibc message received on the host

Reviewer 2: sounds complicated with a few redundant (IMO) terms introduced and some acronyms that better be elaborated because users need to understand each of the parameters in order to determine the values. can't we go like that?
```suggestion
- `binary` is a binary on the chain you are working with (the remote chain)
- `channel-id` is the ID of the ICA's channel on the remote chain's side
- `seq-id` - sequence ID of the IBC message received on the remove chain
```

might be wrong in something, just a brief suggestion for a better understanding

Reviewer 2: also would be useful for a reader to have a hint on how to find these values

> For example `gaiad q interchain-accounts host packet-events channel-736 1 --node https://cosmos-rpc.quickapi.com:443` is the transaction `fund community pool from neutron unclaimed airdrop` on `cosmoshub-4` chain. The node may remove tx from storage in future. You may need to find another one to get details.

Author: patch with error emitted as event introduced just 3weeks ago - https://github.com/cosmos/ibc-go/pull/5541. Not sure there any network with patched ibc released, so there is no realworld example. But i've added command output from local chain (gaiad main)

2. Limit the amount of gas available for sudo handler execution. Out of gas panic will later be captured by `SudoLimitWrapper` and converted into an error.

## Failed interchain txs

Not every interchaintx executes succesfully on a remote network. Some of them fail to execute with errors and then you get ibc acknowledgement with `Error` type. The error is passed into the caller contract via sudo call with SudoMsg::Error [variant](../../../tutorials/cosmwasm_ica.md#ibc-events)

Unfortunately, to avoid the nondeterminism associated with error text generation, the error text is severely truncated by [redact down](https://github.com/cosmos/ibc-go/blob/v7.3.1/modules/apps/27-interchain-accounts/host/ibc_module.go#L115) to the error code without any additional details, before converting into AcknowledgementError.

Find the error text is possible if host chain includes ibc-go v7.2.3+, v7.3.2+, v8.0.1+, v8.1+ which include patch [5541](https://github.com/cosmos/ibc-go/pull/5541)
`<binary> q interchain-accounts host packet-events <channel-id> <seq-id>`

Where:

- `binary` is a binary on the chain you are working with (the remote chain)

> - `seq-id` - sequence ID of the IBC message sent to the remove chain. The seq-id is returned to the contract in the [SubmitTx](./messages.md#response) response

Reviewer 2: ```suggestion
- `seq-id` - sequence ID of the IBC message sent to the remote chain. The seq-id is returned to the contract in the [SubmitTx](./messages.md#response) response
```

- `channel-id` is the ID of the ICA's channel on the remote chain's side. You should know it from registration [procedure](../../../tutorials/cosmwasm-ica.md#2-register-an-interchain-account) via `SudoMsg::OpenAck` from `counterparty_channel_id` field. If you missed it you can always get counterparty channel-id with CLI command `neutrond q ibc channel end <src-port> <src-channel-id>`
- `src-channel-id` is the channel you intechain account associated with.

> - `src-port` is the channel you intechain account associated with.

Reviewer 2: ```suggestion
- `src-port` is the port you interchain account is associated with.
```
